record/record_test.go

```diff
@@ -312,3 +313,137 @@ func TestStaleWriter(t *testing.T) {
 		t.Fatalf("stale write #1: unexpected error: %v", err)
 	}
 }
+
+func TestBasicReadRecovery(t *testing.T) {
+	buf := new(bytes.Buffer)
+	w := NewWriter(buf)
+
+	for i := 0; i < 3; i++ {
+		if wRec, err := w.Next(); err != nil {
+			t.Fatal(err)
+		} else {
```

Comment:
Drop the else and outdent.
https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

wRec, err := w.Next()
if err != nil {
  t.Fatal(err)
}
if _, err = wRec.Write(etc); err != nil {